controller/linodecluster_controller_helpers.go

```diff
@@ -31,12 +30,6 @@ func addMachineToLB(ctx context.Context, clusterScope *scope.ClusterScope) error
 		}
 		return nil
 	}
-	if clusterScope.LinodeCluster.Spec.Network.NodeBalancerID == nil {
```

Comment:
If feel this check is redundant, because the createReconcile explicitly has it own requeueing and sets at the very last. `clusterScope.LinodeCluster.Spec.ControlPlaneEndpoint = clusterv1.APIEndpoint{}`. Ou execution will only reach inside the loop when both LinodeCluster.Spec.Network.NodeBalancerID and clusterScope.LinodeCluster.Spec.Network.ApiserverNodeBalancerConfigID are correctly set.